### PR TITLE
Fix long string decode

### PR DESCRIPTION
**Pull request by wemeetagain**
Fix a few edge-cases in decoding long strings that previously could cause OOM crash.
Found investigating https://github.com/ChainSafe/discv5/issues/64

### CODE REVIEW

**Comment by ryanio** [comment]
thanks @wemeetagain, this looks great.

i noticed the coverage step was failing to push to coveralls due to:

```
[error] "2020-07-10T20:06:25.737Z"  'error from getOptions'
##[error]Error: Command failed: git cat-file -p 3ebc75f2f9e2e2368d93fd9c4d671536713be8de
fatal: Not a valid object name 3ebc75f2f9e2e2368d93fd9c4d671536713be8de
```

which I actually just ran into at web3.js a few days ago. the solution was to update `actions/checkout` to v2 (see [here](https://github.com/coverallsapp/github-action/issues/55) for more). i opened #92 to fix.

**Comment by holgerd77** [comment]
Thanks @wemeetagain, I have merged in #92, this would need a branch update/rebase here for CI to re-run, don't have the permissions to trigger myself.

**Comment by holgerd77** [comment]
(please give an explicit comment statement once you have done, these branch updates always tends to get lost/overlooked otherwise)

**Comment by wemeetagain** [comment]
Rebased off latest!

File: test/dataTypes.spec.ts [hunk 1]
```diff
@@ -5,15 +5,36 @@ const BN = require('bn.js')
 const Buffer = require('buffer').Buffer // needed for karma
 
 describe('invalid rlps', function() {
-  it('should not crash on an invalid rlp', function() {
+  const errCases = [
     // prettier-ignore
```
Comment (ryanio) [inline test/dataTypes.spec.ts:9]: 👀 oh that's neat

**Comment by holgerd77** [comment]
@wemeetagain thanks for fixing, will directly prepare a release.

**Pull request merged**